stlab/channel.hpp

```diff
@@ -352,17 +388,24 @@ struct shared_process : shared_process_receiver<yield_type<T, Arg>>,
     }
 
     void clear_to_send() override {
+        // The downstream sender may trigger this because an other parallel upstream
+        // process may have just passed down a value
+        if (_process_suspend_count == 0)
+            return;
+
         bool do_run = false;
         {
             std::unique_lock<std::mutex> lock(_process_mutex);
+            auto ps = get_process_state(_process);
             --_process_suspend_count; // could be atomic?
             assert(_process_running && "ERROR (sparent) : clear_to_send but not running!");
             if (!_process_suspend_count) {
                 // FIXME (sparent): This is calling the process state ender the lock.
```

Comment:
Comment is now dangling - correct thing is to set a flag that suspend count is 0. The release the lock and do the final checks.